demo_apps/RAG_Chatbot_example/RAG_Chatbot_Example.ipynb

```diff
@@ -0,0 +1,631 @@
+    "        q = Queue()\n",
+    "\n",
+    "        #Update new llama hyperparameters\n",
+    "        llm.inference_server_url = model_selector\n",
+    "        llm.temperature = temperature_selector\n",
+    "        llm.top_p = top_p_selector\n",
+    "        llm.max_new_tokens = max_new_tokens_selector\n",
+    "\n",
+    "        #Async task for streamed chain results wired to callbacks we previously defined, so we don't block the UI\n",
+    "        async def task(prompt):\n",
+    "            ret = await qa_chain.run(prompt, callbacks=[MyStream(q)])\n",
+    "            return ret\n",
+    "\n",
+    "        with start_blocking_portal() as portal:\n",
+    "            portal.start_task_soon(task, user_prompt_message)\n",
+    "            while True:\n",
+    "                next_token = q.get(True)\n",
+    "                if next_token is job_done:\n",
+    "                    messages_history += [{\"role\": \"assistant\", \"content\": bot_message}]\n",
+    "                    return history, messages_history\n",
+    "                bot_message += next_token\n",
+    "                history[-1][1] += next_token\n",
+    "                yield history, messages_history\n",
+    "\n",
+    "    #init the chat history with default system message    \n",
+    "    def init_history(messages_history):\n",
+    "        messages_history = []\n",
+    "        messages_history += [system_message]\n",
+    "        return messages_history\n",
+    "\n",
+    "    #clean up the user input text field\n",
+    "    def input_cleanup():\n",
+    "        return \"\"\n",
+    "\n",
+    "    #when user click Enter and user message are submitted\n",
+    "    user_prompt_message.submit(user, [user_prompt_message, chatbot], [chatbot], queue=False).then(\n",
+    "        bot, [model_selector, temperature_selector, top_p_selector, max_new_tokens_selector, user_prompt_message, chatbot, state], [chatbot, state]\n",
+    "    ).then(input_cleanup, [], [user_prompt_message], queue=False)\n",
+    "\n",
+    "    #when user click the submit button\n",
+    "    submitBtn.click(user, [user_prompt_message, chatbot], [chatbot], queue=False).then(\n",
+    "        bot, [model_selector, temperature_selector, top_p_selector, max_new_tokens_selector, user_prompt_message, chatbot, state], [chatbot, state]\n",
+    "    ).then(input_cleanup, [], [user_prompt_message], queue=False)\n",
+    "    \n",
+    "    #when user click the clear button\n",
+    "    clear.click(lambda: None, None, chatbot, queue=False).success(init_history, [state], [state])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Lastly, we can launch this demo on our localhost. Make sure you have access to the ports.  \n",
+    "In the notebook or a browser with URL http://0.0.0.0:7860 you should see the UI.  \n",
+    "Things to try in the chatbot demo:  \n",
+    "* Specific questions relates Llama 2 get started\n",
```

Comment:
Specific questions relates Llama 2 get started -> Asking specific questions related to the Llama 2 Getting Started Guide